Winleafs.Wpf/Api/Events/SpotifyEventTrigger.cs

```diff
@@ -0,0 +1,47 @@
+﻿using NLog;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace Winleafs.Wpf.Api.Events
+{
+    /// <summary>
+    /// Event trigger that is activated when a Spotify playlist is played
+    /// </summary>
+    public class SpotifyEventTrigger : EventTriggerBase
+    {
+        private readonly EventTriggersCollection _eventTriggersCollection;
+        private readonly string _playlistId;
+
+        public SpotifyEventTrigger(EventTriggersCollection eventTriggersCollection, Models.Models.Scheduling.Triggers.SpotifyEventTrigger spotifyEventTrigger)
+            : base(spotifyEventTrigger.Brightness, spotifyEventTrigger.EffectName, spotifyEventTrigger.Priority)
+        {
+            _eventTriggersCollection = eventTriggersCollection;
+            _playlistId = spotifyEventTrigger.PlaylistId;
+
+            var processCheckTimer = new Timer(60000);
+            processCheckTimer.Elapsed += CheckProcess;
+            processCheckTimer.AutoReset = true;
+            processCheckTimer.Start();
+        }
+
+        private void CheckProcess(object source, ElapsedEventArgs e)
+        {
+            Task.Run(() => CheckPlaylistAsync());
+        }
+
+        /// <summary>
+        /// Checks if a playlist is being played in Spotify then execute TryStartEffect(), else stop the effect
+        /// </summary>
+        private async Task CheckPlaylistAsync()
```

Comment:
Second comment on this, couldn't this work with an event? Let the SpotifyTimer poll if there is a change in playlist, if so trigger an event and set the right effect?

Reply:
Possible in theory, but I opted for separation of responsibilities. The SpotifyTimer is repsonsible for getting the current playing palylist ID and the triggers individually check if they should be active.

Reply:
Fair enough, would invalidate the structure already at hand.